### PR TITLE
Use `logging.pattern.level` recommended by Spring

### PR DESCRIPTION
**Pull request by neiser**
The Spring Boot documentation recommends using `logging.pattern.level` in a "Tip" section in https://docs.spring.io/spring-boot/docs/current/reference/html/features.html#features.logging.custom-log-configuration

We should follow this best practice.

### CODE REVIEW

File: docs/logger-mdc-instrumentation.md [hunk 1]
```diff
@@ -19,11 +19,12 @@ event's MDC copy:
   (same as `Span.current().getSpanContext().getTraceFlags().asHex()`).
 
 Those three pieces of information can be included in log statements produced by the logging library
-by specifying them in the pattern/format. Example for Spring Boot configuration (which uses
-logback):
+by specifying them in the pattern/format. 
 
+Example for Spring Boot configuration, which uses
+logback and the [last tip of this section](https://docs.spring.io/spring-boot/docs/current/reference/html/features.html#features.logging.custom-log-configuration):
```
Comment (trask) [inline docs/logger-mdc-instrumentation.md:25]: I had not seen this tip before, so it took me a while to realize that `level` was not a mistake 😄

```suggestion
Tip: for Spring Boot configuration which uses logback, you can add MDC to log lines by overriding only the logging.pattern.level:
```

File: docs/logger-mdc-instrumentation.md [hunk 2]
```diff
@@ -19,11 +19,12 @@ event's MDC copy:
...
 ```properties
-logging.pattern.console = %d{yyyy-MM-dd HH:mm:ss} - %logger{36} - %msg trace_id=%X{trace_id} span_id=%X{span_id} trace_flags=%X{trace_flags} %n
+logging.pattern.level = %prefix(%mdc{trace_id:-0}) %prefix(%mdc{span_id:-0}) %prefix(%mdc{trace_flags:-0}) %5p
```
Comment (trask) [inline docs/logger-mdc-instrumentation.md:27]: what is the purpose of the default?

Comment (neiser) [reply docs/logger-mdc-instrumentation.md:26]: Ah, well, we can omit it!

File: docs/logger-mdc-instrumentation.md [hunk 3]
```diff
@@ -19,11 +19,11 @@ event's MDC copy:
...
 
 Those three pieces of information can be included in log statements produced by the logging library
-by specifying them in the pattern/format. Example for Spring Boot configuration (which uses
-logback):
+by specifying them in the pattern/format. 
 
+Tip: for Spring Boot configuration which uses logback, you can add MDC to log lines by overriding only the logging.pattern.level:
 ```properties
-logging.pattern.console = %d{yyyy-MM-dd HH:mm:ss} - %logger{36} - %msg trace_id=%X{trace_id} span_id=%X{span_id} trace_flags=%X{trace_flags} %n
+logging.pattern.level = %prefix(%mdc{trace_id}) %prefix(%mdc{span_id}) %prefix(%mdc{trace_flags}) %5p
```
Comment (trask) [inline docs/logger-mdc-instrumentation.md:26]: I had to look up what `%prefix` did, it looks cool, but I think maybe better to keep the example here simple to understand

Comment (neiser) [reply docs/logger-mdc-instrumentation.md:26]: Hm, I'd say that also before the change, one would need some understanding of logback and look up what `%X` actually means. I've followed your advice though, but kept the more explicit `%mdc`.

File: docs/logger-mdc-instrumentation.md [hunk 4]
```diff
@@ -19,11 +19,11 @@ event's MDC copy:
   (same as `Span.current().getSpanContext().getTraceFlags().asHex()`).
 
 Those three pieces of information can be included in log statements produced by the logging library
-by specifying them in the pattern/format. Example for Spring Boot configuration (which uses
-logback):
+by specifying them in the pattern/format. 
 
+Tip: for Spring Boot configuration which uses logback, you can add MDC to log lines by overriding only the logging.pattern.level:
```
Comment (mateuszrzeszutek) [inline docs/logger-mdc-instrumentation.md:24]: Nit
```suggestion
Tip: for Spring Boot configuration which uses logback, you can add MDC to log lines by overriding only the `logging.pattern.level`:
```

**Review by mateuszrzeszutek — APPROVED** [review]

**Review by trask — APPROVED** [review]
thx @neiser!

**Pull request merged**